src/GafferScene/OpenGLShader.cpp

```diff
@@ -136,7 +144,7 @@ void OpenGLShader::loadShader( const std::string &shaderName, bool keepExistingV
 		if( plug )
 		{
 			plug->setFlags( Plug::Dynamic, true );
-			parametersPlug()->addChild( plug );
+			PlugAlgo::replacePlug( parametersPlug(), plug );
```

Comment:
While this works, it will result in unnecessary churn when reloading, because we're unconditionally replacing the existing plug rather than reusing it where possible. Ideally a reload when the shader hasn't changed will do nothing at all. Seems like we might be able to achieve this fairly simply something along these lines?

```
const Plug *existingPlug = parametersPlug()->getChild<Plug>( *it ) )
const TypeId existingType = existingPlug ? existingPlug->typeId() : IECore::InvalidTypeId;
switch( parameter->type )
 	{
 		case GL_BOOL :
 			plug = existingType != BoolPlugTypeId ? new BoolPlug( *it ) : NULL;
 			break;
                ...
```